### PR TITLE
PCC16 nonickid

### PR DESCRIPTION
**Pull request by nonickid**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): [4]
Estimated time spent (hours): [4]
Completed (yes/no): [Yes]
I stretched my coding skills (if yes what did you learn?): [Improve my knowledge and skill to use API requests in Python]
Other feedback (what can we improve?): []

### CODE REVIEW

File: 16/nonickid/top_apple_songs.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        print("Error: Please check help to provide proper values!")
+        sys.exit(1)
+ 
+    songs = get_data(url.format(args.code.lower(), args.limit))
+
+    print()
+    print_banner(countries[args.code.upper()], songs['feed']['updated'])
+
+    for i, song in enumerate(songs['feed']['results'], start=1):
+        print(f"{i:>4} | {song['name'][:50]:<50} | {song['artistName'][:50]:<50} | {song['genres'][0]['name']}")
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: break this out over multiple lines

File: 16/nonickid/top_apple_songs.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    if args.code.upper() not in countries or args.limit < 1 or args.limit > 100:
+        parser.print_help()
+        print("Error: Please check help to provide proper values!")
+        sys.exit(1)
+ 
+    songs = get_data(url.format(args.code.lower(), args.limit))
+
+    print()
+    print_banner(countries[args.code.upper()], songs['feed']['updated'])
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: what if `songs` is `None`?

File: 16/nonickid/top_apple_songs.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    # Apple's API for music library
+    url = "https://rss.itunes.apple.com/api/v1/{}/apple-music/top-songs/all/{}/explicit.json"
+    # Country code API
+    url_country = "http://country.io/names.json"
+
+    countries = get_data(url_country)
+
+    if args.code.upper() not in countries or args.limit < 1 or args.limit > 100:
+        parser.print_help()
+        print("Error: Please check help to provide proper values!")
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: consider `logging`

File: 16/nonickid/top_apple_songs.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    banner = f"Apple's Music Top Songs for {code}"
+    lastUpdate = f"Last list update {update}"
+
+    print(f"{banner:^130}")
+    print(f"{lastUpdate:^130}")
+    print("-"*130)
+    print("{0:>2} | {1:<50} | {2:<50} | {3}".format("Rank", "Song", "Artist", "Genres"))
+    print("-"*130)
+
+def main():
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: I think you can break this down in various smaller functions

File: 16/nonickid/top_apple_songs.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    parser = argparse.ArgumentParser(description="Display Apple's Music Top Songs")
+    parser.add_argument('-c', '--code', required=True, help='Country code (US, DE, AR, PL e.g)')
+    parser.add_argument('-l', '--limit', type=int, required=True, help='Number of songs to display (1-100)')
+    args = parser.parse_args()
+
+    # Apple's API for music library
+    url = "https://rss.itunes.apple.com/api/v1/{}/apple-music/top-songs/all/{}/explicit.json"
+    # Country code API
+    url_country = "http://country.io/names.json"
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: use a constant

File: 16/nonickid/top_apple_songs.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    print(f"{banner:^130}")
+    print(f"{lastUpdate:^130}")
+    print("-"*130)
+    print("{0:>2} | {1:<50} | {2:<50} | {3}".format("Rank", "Song", "Artist", "Genres"))
+    print("-"*130)
+
+def main():
+
+    parser = argparse.ArgumentParser(description="Display Apple's Music Top Songs")
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: nice!

File: 16/nonickid/top_apple_songs.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import json
+import collections
+import argparse
+import sys
+
+
+def get_data(url):
+    resp = requests.get(url)
+    if resp.status_code == 200:
+        return json.loads(resp.text)
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: `requests` has a json method, so you could return `resp.json()`

File: 16/nonickid/top_apple_songs.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
+# Simple program to received Apple Music Top Songs based on country's code provided
+# To received Top Songs list Apple's API is used
+# To received Country list country.io API is used. The data is used to map country code to country name
+import requests
```
Comment (bbelderbos) [inline 16/nonickid/top_apple_songs.py]: I usually split the modules as per pep8: first stdlib modules, blank line, external modules

File: 16/nonickid/requirments.txt [hunk 9]
```diff
@@ -0,0 +1 @@
+requests==2.22.0
```
Comment (bbelderbos) [inline 16/nonickid/requirments.txt]: nice you included requirements

**Comment by bbelderbos** [comment]
Nice practice Adam!

**Comment by nonickid** [comment]
@bbelderbos Thank you so much for all valuable comments. I have taken notes and will use in my further coding.

**Pull request merged**